cfg4k-core/src/test/kotlin/com/jdiazcano/cfg4k/TimedReloadStrategyTest.kt

```diff
@@ -78,8 +81,47 @@ nested.a=reloaded nestedb
             )
             checkProvider(overrideFile, provider, text, true)
         }
-    }
 
+        it("error on reload doesn't prevent further reload attempts") {
+            var reloadCounter = 0
+            val provider = reloadTestProvider {
+                // Every second reload throws an exception
+                if (reloadCounter++ % 2 == 0) throw Exception("simulate reload failure")
+            }
+            val reloadStrategy = TimedReloadStrategy(100, TimeUnit.MILLISECONDS)
+
+            reloadStrategy.register(provider)
+            Thread.sleep(1000)
+            reloadStrategy.deregister(provider)
+
+            // Expected value is low enough to avoid test flakiness
+            reloadCounter.should.be.least(5)
+        }
+
+        it("reload strategy can be reused by multiple providers") {
+            var reloadCounter1 = 0
+            val provider1 = reloadTestProvider { reloadCounter1 ++ }
+            var reloadCounter2 = 0
+            val provider2 = reloadTestProvider { reloadCounter2 ++ }
+            val reloadStrategy = TimedReloadStrategy(10, TimeUnit.MILLISECONDS)
+
+            reloadStrategy.register(provider1)
+            reloadStrategy.register(provider2)
+            Thread.sleep(100)
+            reloadStrategy.deregister(provider1)
+            val lastSeenReloadCounter1 = reloadCounter1
+            val lastSeenReloadCounter2 = reloadCounter2
+            Thread.sleep(100)
+            reloadStrategy.deregister(provider2)
+            val finishedReloadCounter2 = reloadCounter2
```

Comment:
This variable doesn't seem to be used. Is this a miss or can be removed?

Reply:
Yes, a leftover. Interesting, it didn't appear as not used but it's obviously not used 🤔